packages/cells/src/model.ts

```diff
@@ -681,9 +686,35 @@ export class CodeCellModel extends CellModel implements ICodeCellModel {
     this.modelDB.setValue('executionCount', newValue || null);
   }
 
+  /**
+   * Whether the cell is dirty or not.
+   *
+   * A cell is dirty if it is output is not empty and does not
+   * result of the input code execution.
+   */
+  get isDirty(): boolean {
+    // Test could be done dynamically with this._executedCode
+    // but for performance reason, the diff status is stored in a boolean.
+    return this._isDirty;
+  }
+  set isDirty(v: boolean) {
```

Comment:
Do we need this setter to be public? Do we expect users of the model to be able to set the flag? Or could this be a private method?

Reply:
We need commands like split-cell to be able to set the dirty flag, so I am tempted to say yes, but I might not have the understanding of the whole thing.

Reply:
Right, but will these commands manipulate the dirty state directly on the model?

Reply:
That's what I understood yes.

The annoying thing is that those commands manipulate the JSON model of the cell, the one that relates to what is actually loaded and saved in the Notebook file, at least that is what I understood, please correct me if I am wrong. So we might need the dirty state to be part of the official notebook format before we can implement this correctly.

Reply:
Ah ok. I was searching the code from this branch for when the dirty state is set with `.isDirty = `, and it seemed to be coming from the model itself only. Which suggested having the setter be public might not be needed?

Because then we could have changed the attribute on the interface to be: `readonly isDirty: boolean;`

Reply:
Sure. It's fine making it readonly. We can change it later if needed :)